server/sonar-server/src/main/java/org/sonar/server/email/ws/UpdateAction.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.email.ws;
+
+import javax.annotation.Nullable;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.property.PropertyDto;
+import org.sonar.server.user.UserSession;
+
+import static org.sonar.api.config.EmailSettings.FROM;
+import static org.sonar.api.config.EmailSettings.FROM_DEFAULT;
+import static org.sonar.api.config.EmailSettings.PREFIX;
+import static org.sonar.api.config.EmailSettings.PREFIX_DEFAULT;
+import static org.sonar.api.config.EmailSettings.SMTP_HOST;
+import static org.sonar.api.config.EmailSettings.SMTP_PASSWORD;
+import static org.sonar.api.config.EmailSettings.SMTP_PORT;
+import static org.sonar.api.config.EmailSettings.SMTP_PORT_DEFAULT;
+import static org.sonar.api.config.EmailSettings.SMTP_SECURE_CONNECTION;
+import static org.sonar.api.config.EmailSettings.SMTP_USERNAME;
+
+public class UpdateAction implements EmailsWsAction {
+
+  private static final String PARAM_HOST = "host";
+  private static final String PARAM_PORT = "port";
+  private static final String PARAM_SECURE = "secure";
+  private static final String PARAM_USERNAME = "username";
+  private static final String PARAM_PASSWORD = "password";
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2068)